File: leetcode-submissions/202_Happy_Number.ts
function isHappy(n: number): boolean {
    const set:Set<number>= new Set();
    while(true){
        let sum :number =0;
        while(n>0){
            let lastDigit:number = n%10;
            let sq:number = lastDigit*lastDigit;
            sum+=sq;
            n = Math.floor(n / 10);

        }
        if(set.has(sum)) return false;
        set.add(sum);
        if(sum===1) return true;
        n=sum;
    }
    return true;
};

File: leetcode-submissions/202_Happy_Number.cpp
class Solution {
public:
    bool isHappy(int n) {
       long long int sum=0;
        set<int>s;
        while(1){
            sum=0;
            while(n>0){
                int lastdigit=n%10;
                sum=sum+(lastdigit*lastdigit);
                n=n/10;

            }
            if(sum==1){
                return true;
            }
            if(s.count(sum)){
                return false;
            }else{
                s.insert(sum);
            }
            n=sum;
            
        }
        return true;
    }
};